EmployeeList/ViewModels/EmployeeListViewModel.swift

```diff
@@ -0,0 +1,87 @@
+//
+//  EmployeeListViewModel.swift
+//  EmployeeList
+//
+//  Created by Wang Siman on 1/29/21.
+//  Copyright © 2021 None. All rights reserved.
+//
+
+import Foundation
+import Combine
+import SwiftUI
+
+final class EmployeeListViewModel: ObservableObject {
+    private var cancellables: [AnyCancellable] = []
+    
+    private let onAppearSubject = PassthroughSubject<Void, Error>()
+    
+    private let employeesService: APIServiceType
+    
+    // Publish values on-demand by calling the send() method
+    func update() {
```

Comment:
I'm not sure I follow this one. Why aren't you using objectWillChange to indicate changes?

Reply:
Calling objectWillChange.send() in views is definitely fancier than my approach! To be honset, I was not familiar with objectWillChange at that moment.

I also consider the situation where the employee list is very large such that I need to fetch a few employees with a pagniated endpoint. In this case, I can specify the behavior of viewModel to load more data from the next page when the user reaches the end by passing arguments to `upate()`.

For instance,
```swift
enum Action {
    case appear
    case loadMore
}

func update(action: Action, currentItem: Employee) {
    switch Action {
    case .appear: { do something } // used when employees are in a single JSON
    case .loadMore: { do something } // used when employees are divided into multiple JSON
    }
}

```